include/flashinfer/attention/cascade.cuh

```diff
@@ -336,20 +340,22 @@ __global__ void MergeStatesLargeNumIndexSetsKernel(DTypeIn* __restrict__ V, floa
  * \param indptr The start offsets of each position in the variable length array.
  * \param v_merged The merged v of index sets union. (n, h, d)
  * \param s_merged The merged logsumexp value of index sets union. (n, h)
+ * \param max_seq_len The maximum sequence length supported by the kernel.
+ * \param seq_len_ptr The current sequence length (number of positions populated in indptr).
  * \param num_heads The number of heads of v.
  * \param head_dim The dimension of each head.
  * \note s are logsumexp values with base 2.
  */
 template <uint32_t vec_size, uint32_t bdx, uint32_t bdy, uint32_t num_smem_stages, typename DTypeIn,
           typename DTypeO, typename IdType>
-__global__ void PersistentVariableLengthMergeStatesKernel(DTypeIn* __restrict__ V,
-                                                          float* __restrict__ S, IdType* indptr,
-                                                          DTypeO* __restrict__ v_merged,
-                                                          float* __restrict__ s_merged,
-                                                          uint32_t seq_len, uint32_t num_heads) {
+__global__ void PersistentVariableLengthMergeStatesKernel(
+    DTypeIn* __restrict__ V, float* __restrict__ S, IdType* indptr, DTypeO* __restrict__ v_merged,
+    float* __restrict__ s_merged, uint32_t max_seq_len, uint32_t* __restrict__ seq_len_ptr,
```

Comment:
another option is to make `seq_len` a cuda array with length 1 and always read `seq_len[0]`'s value inside kernels, but currently I think having another `max_seq_len` argument is okay.